flang-rt/lib/runtime/extensions.cpp

```diff
@@ -268,5 +269,8 @@ void FORTRAN_PROCEDURE_NAME(qsort)(int *array, int *len, int *isize,
   qsort(array, *len, *isize, compar);
 }
 
+// PERROR(STRING)
+void RTNAME(Perror)(const char *str) { perror(str); }
```

Comment:
This code assumes that the Fortran character value is NUL-terminated, and ignores its length.  The GNU documentation for `PERROR` does not require NUL termination.